libs/wingsdk/src/core/app.ts

```diff
@@ -225,8 +213,13 @@ export abstract class App extends Construct implements IApp {
     id: string,
     ...args: any[]
   ): any {
-    // next delegate to "tryNew", which will allow derived classes to inject
-    const instance = this.tryNew(fqn, scope, id, ...args);
+    // uses the Back Reference for the platform manager to determine which what type to create
+    const instance = this._platformManager?._tryNewInstance(
```

Comment:
If `this._platformManager` is undefined, then the method will always throw an error. Can we change `this._platformManager` to be a required field?